pmd-cataloger/src/main/java/sfdc/sfdx/scanner/SfdxScannerException.java

```diff
@@ -13,33 +13,36 @@
 public class SfdxScannerException extends RuntimeException {
 
   private final EventKey eventKey;
-  private final List<String> args;
+  private final String[] args;
 
-  public SfdxScannerException(EventKey eventKey, String[] args) {
-    this(eventKey, args, null);
+  public SfdxScannerException(EventKey eventKey, String... args) {
+    this(eventKey, null, args);
   }
 
-  public SfdxScannerException(EventKey eventKey, String[] args, Throwable throwable) {
+  public SfdxScannerException(EventKey eventKey, Throwable throwable, String... args) {
     super(throwable);
 
-    // Confirm that the correct number of arguments for the message has been provided
-    // If this fails, this would be a developer error
-    assert (eventKey.getArgCount() == args.length);
-
     this.eventKey = eventKey;
-    this.args = Arrays.asList(args);
+    this.args = args;
   }
 
   public EventKey getEventKey() {
     return eventKey;
   }
 
-  public List<String> getArgs() {
+  public String[] getArgs() {
     return args;
   }
 
   public String getFullStacktrace() {
     return Throwables.getStackTraceAsString(this).replace("\\n", " | ");
   }
 
+  @Override
+  public String toString() {
+    return "SfdxScannerException{" +
```

Comment:
Do you use the `toString()` method when building the message that the TypeScript will read? If so, does starting the output with "SfdxScannerException" mess anything up?

Reply:
No, `toString()` is just a matter of best practice so that we get useful information while debugging code.
For Typescript to read, this exception is converted into an instance of `Message.java` so that the fields are the same. Have a look at the changes in`SfdxMessager.java`.